devtools/datasette/publish.py

```diff
@@ -138,48 +150,48 @@ def deploy_datasette(
 
     python publish.py --fly -- --build-only
     """
+    logging.info(f"Deploying to {deploy.upper()}...")
     pudl_output = PudlPaths().pudl_output
 
-    pudl_output = PudlPaths().pudl_output
-    metadata_yml = DatasetteMetadata.from_data_source_ids(pudl_output).to_yaml()
-
     all_databases = (
         ["pudl.sqlite"]
         + sorted(str(p.name) for p in pudl_output.glob("ferc*.sqlite"))
         + ["censusdp1tract.sqlite"]
     )
     databases = list(only_databases if only_databases else all_databases)
 
-    # Make sure we have the expected metadata for databases
-    # headed to deployment.
-    if deploy != "metadata":
-        check_tables_have_metadata(metadata_yml, databases)
-    if deploy in {"production", "staging"}:
-        fly_dir = Path(__file__).parent.absolute() / "fly"
-        logging.info(f"Deploying {deploy} to fly.io...")
-        docker_path = fly_dir / "Dockerfile"
-        inspect_path = fly_dir / "inspect-data.json"
-        metadata_path = fly_dir / "metadata.yml"
-
-        logging.info(f"Inspecting DBs for datasette: {databases}...")
-        inspect_output = inspect_data(databases, pudl_output)
-        with inspect_path.open("w") as f:
-            f.write(json.dumps(inspect_output))
-
-        logging.info(f"Writing Datasette metadata to: {metadata_path}")
-        with metadata_path.open("w") as f:
-            f.write(metadata_yml)
-
-        logging.info("Writing Dockerfile...")
-        with docker_path.open("w") as f:
-            f.write(make_dockerfile(databases))
-
-        logging.info(f"Compressing {databases} and putting into docker context...")
-        check_call(
-            ["tar", "-a", "-czvf", fly_dir / "all_dbs.tar.zst"] + databases,  # noqa: S603
-            cwd=pudl_output,
-        )
+    fly_dir = Path(__file__).parent.absolute() / "fly"
+    docker_path = fly_dir / "Dockerfile"
+    inspect_path = fly_dir / "inspect-data.json"
+    metadata_path = fly_dir / "metadata.yml"
+    metadata_yml = DatasetteMetadata.from_data_source_ids(pudl_output).to_yaml()
+
+    logging.info(f"Writing Datasette metadata to: {metadata_path}")
+    with metadata_path.open("w") as f:
+        f.write(metadata_yml)
+    check_tables_have_metadata(metadata_yml, databases)
+
+    if deploy == "metadata":
+        logging.info("Only writing metadata. Aborting now.")
```

Comment:
You maybe forgot to abort the script here?

Reply:
Sure did, fixing! Fortunately we only *actually deploy* if we actively choose a valid deployment target :sweat_smile: